src/main/java/com/synopsys/integration/detect/workflow/bdio/AggregateBdioCreator.java

```diff
@@ -129,13 +130,19 @@ public AggregateBdioCreator(final Bdio2Factory bdio2Factory, final SimpleBdioFac
         }
     }
 
-    private DependencyGraph createAggregateDependencyGraph(final File sourcePath, final List<DetectCodeLocation> codeLocations) {
+    private DependencyGraph createAggregateDependencyGraph(final File sourcePath, final List<DetectCodeLocation> codeLocations, final AggregateMode aggregateMode) throws DetectUserFriendlyException {
         final MutableDependencyGraph aggregateDependencyGraph = simpleBdioFactory.createMutableDependencyGraph();
 
         for (final DetectCodeLocation detectCodeLocation : codeLocations) {
-            final Dependency codeLocationDependency = createAggregateDependency(sourcePath, detectCodeLocation);
-            aggregateDependencyGraph.addChildrenToRoot(codeLocationDependency);
-            aggregateDependencyGraph.addGraphAsChildrenToParent(codeLocationDependency, detectCodeLocation.getDependencyGraph());
+            if (aggregateMode.equals(AggregateMode.DIRECT)) {
+                aggregateDependencyGraph.addGraphAsChildrenToRoot(detectCodeLocation.getDependencyGraph());
+            } else if (aggregateMode.equals(AggregateMode.TRANSITIVE)) {
+                final Dependency codeLocationDependency = createAggregateDependency(sourcePath, detectCodeLocation);
+                aggregateDependencyGraph.addChildrenToRoot(codeLocationDependency);
+                aggregateDependencyGraph.addGraphAsChildrenToParent(codeLocationDependency, detectCodeLocation.getDependencyGraph());
+            } else {
+                throw new DetectUserFriendlyException("Did not specify aggregation mode via detect.bom.aggregate.mode, will not aggregate at this time.", ExitCodeType.FAILURE_GENERAL_ERROR);
```

Comment:
How about the following?

`throw new DetectUserFriendlyException(String.format("The %s property was set to an unsupported aggregation mode, will not aggregate at this time.", DetectProperty.DETECT_BOM_AGGREGATE_MODE.getPropertyKey()), ExitCodeType.FAILURE_GENERAL_ERROR);`